### PR TITLE
fix stable pkg proj versioning support

### PR DESCRIPTION
**Pull request by Anipik**
Fixes https://github.com/dotnet/runtime/issues/57982

### CODE REVIEW

**Review by mmitche — APPROVED** [review]

File: src/coreclr/.nuget/Directory.Build.props [hunk 1]
```diff
@@ -18,8 +18,13 @@
...
-    <PackageVersion Condition="'$(PackageVersion)' == ''">6.0.0</PackageVersion>
-    <StableVersion Condition="'$(StabilizePackageVersion)' == 'true' and '$(StableVersion)' == ''">$(PackageVersion)</StableVersion>
+    <PackageVersion Condition="'$(PackageVersion)' == ''">$(ProductVersion)</PackageVersion>
+    <StableVersion Condition="'$(StabilizePackageVersion)' == 'true' and '$(StableVersion)' == '' and '$(PreReleaseVersionLabel)' != 'servicing'">$(PackageVersion)</StableVersion>
+    <StableVersion Condition="($(MSBuildProjectName.Contains('Private')) or $(MSBuildProjectName.Contains('Experimental')) or $(MSBuildProjectName.Contains('Transport'))) and '$(MSBuildProjectExtension)' == '.pkgproj'"></StableVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(PreReleaseVersionLabel)' == 'servicing'">
+    <IsShipping>false</IsShipping>
   </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (Anipik) [reply src/coreclr/.nuget/Directory.Build.props:28]: they will get shipped internally

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/coreclr/.nuget/Directory.Build.props:28]: Why do we change the definition of what's shipping publicly vs internally based on if we the repo is in servicing mode or not?

[the remark this answers is not in the document]
Comment (mmitche) [reply src/coreclr/.nuget/Directory.Build.props:28]: What packages are these specifically? IsShipping generally controls whether something goes to NuGet. Are they supposed to go to NuGet in non-servicing?

[the remark this answers is not in the document]
Comment (Anipik) [reply src/coreclr/.nuget/Directory.Build.props:28]: > Why do we change the definition of what's shipping publicly vs internally based on if the repo is in servicing mode?

basically following the same convention guidelines from the coreclr team as we did for 5.0.0

> What packages are these specifically? IsShipping generally controls whether something goes to NuGet. Are they supposed to go to NuGet in non-servicing?

yes they are not supposed to go to nuget in non-servicing

[the remark this answers is not in the document]
Comment (mmitche) [reply src/coreclr/.nuget/Directory.Build.props:28]: But they do go in rtm? What package set is this specifically?

[the remark this answers is not in the document]
Comment (Anipik) [reply src/coreclr/.nuget/Directory.Build.props:28]: yes they go to nuget in rtm, these https://github.com/dotnet/runtime/tree/main/src/coreclr/.nuget
```
Microsoft.ILVerification
Microsoft.NETCore.ILAsm
Microsoft.NETCore.ILDAsm
Microsoft.NETCore.TestHost
dotnet-ilverify
```

**Pull request merged**